### PR TITLE
KES dependency upgrade

### PR DESCRIPTION
**Pull request by Alevsk**
## Description

- Upgrading KES dependency to v.0.18.0
- Fixing incompatibility issue when checking for errors during KES key creation

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can you run `go mod tidy -compat=1.17`  locally @Alevsk

File: cmd/common-main.go [hunk 1]
```diff
@@ -819,8 +819,9 @@ func handleCommonEnvVars() {
 		// We check that the default key ID exists or try to create it otherwise.
 		// This implicitly checks that we can communicate to KES. We don't treat
 		// a policy error as failure condition since MinIO may not have the permission
+		// a policy error as failure condition since MinIO may not have the permission
```
Comment (harshavardhana) [inline cmd/common-main.go]: both lines are the same we only need one.

File: internal/kms/kes.go [hunk 2]
```diff
@@ -140,3 +141,9 @@ func (c *kesClient) DecryptKey(keyID string, ciphertext []byte, ctx Context) ([]
 	}
 	return c.client.Decrypt(context.Background(), keyID, ciphertext, ctxBytes)
 }
+
+func KeyExists(err error) bool {
+	// ErrPrevKESKeyExists will be used to maintain compatibility with KES versions older than v0.18.0
+	ErrPrevKESKeyExists := kes.NewError(http.StatusBadRequest, "key does already exist")
```
Comment (harshavardhana) [inline internal/kms/kes.go]:
```suggestion
	legacyKeyExists := kes.NewError(http.StatusBadRequest, "key does already exist")
```

File: internal/kms/kes.go [hunk 3]
```diff
@@ -140,3 +141,9 @@ func (c *kesClient) DecryptKey(keyID string, ciphertext []byte, ctx Context) ([]
...
+	return errors.Is(err, kes.ErrKeyExists) || errors.Is(err, ErrPrevKESKeyExists)
```
Comment (harshavardhana) [inline internal/kms/kes.go]:
```suggestion
	return errors.Is(err, kes.ErrKeyExists) || errors.Is(err, legacyKeyExists)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-handler-utils.go [hunk 4]
```diff
@@ -20,9 +20,9 @@ package cmd
 import (
 	"context"
 	"errors"
+	"github.com/minio/minio/internal/kms"
```
Comment (harshavardhana) [inline cmd/admin-handler-utils.go]: this import should be close the other dependencies perhaps you should run `goimports -w`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-handler-utils.go (hunk 4 above)
Comment (Alevsk) [reply cmd/admin-handler-utils.go]: ran goimports, done

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by dvaldivia — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**